GEECS-PythonAPI/geecs_python_api/controls/data_acquisition/multiscan_analysis.py

```diff
@@ -0,0 +1,294 @@
+"""
+
+Multi-scan Analyzer
+
+Contributors:
+Kyle Jensen, kjensen@lbl.gov
+
+"""
+# =============================================================================
+# %% imports
+import os
+import glob
+import cv2
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+from geecs_python_api.controls.data_acquisition.data_acquisition import DataInterface
+from geecs_python_api.controls.data_acquisition.scan_analysis import VisaEBeamAnalysis
+from image_analysis.utils import read_imaq_image
+
+# =============================================================================
+# %% classes
+
+class MultiScanAnalysis():
+    """
+    Intended to be a parent class for generalized multi-scan analysis.
+    Not sure how to generalize at this point.
+    """
+
+    def __init__(self):
+        self.data_interface = DataInterface()
+
+    def parse_scans(self, input_scans):
+        """
+        Parse input scans and organize.
+        """
+
+        # parse input
+        scan_numbers = list(input_scans['scan_num'])
+        scan_years = ([input_scans['year']] * len(scan_numbers)
+                      if isinstance(input_scans['year'], str)
+                      else list(input_scans['year']))
+        scan_months = ([input_scans['month']] * len(scan_numbers)
+                      if isinstance(input_scans['month'], str)
+                      else list(input_scans['month']))
+        scan_days = ([input_scans['day']] * len(scan_numbers)
+                      if isinstance(input_scans['day'], str)
+                      else list(input_scans['day']))
+
+        # check compatibility of scan dates
+        if (len(scan_years) != len(scan_numbers)
+            or len(scan_months) != len(scan_numbers)
+            or len(scan_days) != len(scan_numbers)):
+            raise Exception("Error: Incompatible dates specifications.")
```

Comment:
This formatting of `scan_years`, scan_months`, and `scan_days` seems a little complex at first glance, but looking at an example down on line 279 looks pretty reasonable.  I suppose here you have the option to define a list of scans spanning multiple days/months/years to fully customize your dataset rather than assuming just a single day.  So long as the input formatting is clear than should be fine I think